plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -403,12 +403,20 @@ compileExpr e = withContextM 2 (sdToTxt $ "Compiling expr:" GHC.<+> GHC.ppr e) $
     CompileContext {ccScopes=stack,ccBuiltinNameInfo=nameInfo} <- ask
 
     -- TODO: Maybe share this to avoid repeated lookups. Probably cheap, though.
-    (stringTyName, sbsName) <- case (Map.lookup ''Builtins.String nameInfo, Map.lookup 'String.stringToBuiltinString nameInfo) of
-        (Just t1, Just t2) -> pure $ (GHC.getName t1, GHC.getName t2)
-        _                  -> throwPlain $ CompilationError "No info for String builtin"
+    (stringTyName, sbsName, bsTyName) <- case
+        ( Map.lookup ''Builtins.String nameInfo
+        , Map.lookup 'String.stringToBuiltinString nameInfo
+        , Map.lookup ''Builtins.ByteString nameInfo
+        ) of
+            (Just t1, Just t2, Just t3) -> pure $ (GHC.getName t1, GHC.getName t2, GHC.getName t3)
+            _                           -> throwPlain $ CompilationError "No info for String builtin"
 
     let top = NE.head stack
     case e of
+        -- 'fromString' invocation at the builtin ByteString type
+        (strip -> GHC.Var (GHC.idDetails -> GHC.ClassOpId cls)) `GHC.App` GHC.Type (GHC.tyConAppTyCon_maybe -> Just tc) `GHC.App` _ `GHC.App` (strip -> stringExprContent -> Just bs)
+            | GHC.getName cls == GHC.isStringClassName, GHC.getName tc == bsTyName -> do
+                pure $ PIR.Constant () $ PLC.someValue bs
         -- See Note [String literals]
```

Comment:
This note also applies to the new code, maybe put it above both, they're separated from the rest already so I think that's clear.